core/src/main/java/org/apache/calcite/sql/fun/SqlCoalesceFunction.java

```diff
@@ -29,12 +29,21 @@
 import org.apache.calcite.sql.validate.SqlValidator;
 import org.apache.calcite.util.Util;
 
+import com.google.common.annotations.VisibleForTesting;
+
 import java.util.List;
 
 /**
  * The <code>COALESCE</code> function.
  */
 public class SqlCoalesceFunction extends SqlFunction {
+
+  /**
+   * Only used for testing.
+   */
+  @VisibleForTesting
+  public static boolean enableRewriteCall = true;
```

Comment:
I would prefer if we didn't introduce new production code just for tests even if it is tagged. I guess you can reproduce the problem by creating explicitly a query using the `RelBuilder` and execute it afterwards (see examples in `EnumerableRepeatUnionTest`). An alternative would be to create a test with a custom validator that does not rewrite calls.

Reply:
Thanks @zabetak It's very good for me :)
I refined by your comment ~